pyquil/quilbase.py

```diff
@@ -57,6 +57,28 @@ def _extract_qubit_index(qubit, index=True):
     return qubit.index
 
 
+def _format_qubit_str(qubit):
+    if isinstance(qubit, QubitPlaceholder):
+        return "{%s}" % str(qubit)
```

Comment:
Have you considered using fstringss instead? e.g. `f"{{qubit}}" `

just a small nit

Reply:
Unfortunately f-strings are only in Python 3.6+. However, we should be using .format() instead https://pyformat.info/

Reply:
@stevenheidel Oh, I thought we were Python 3 here. Do we just want to support all Python 3s?

Reply:
We would like to support both Python 3.5 and Python 3.6. That means we can't use the following things:
- f-strings
- underscores in numeric literals
- dataclasses

FWIW qiskit supports Python 3.5 and 3.6. I think this is due to older versions of Ubuntu? I'm not sure, maybe @mpharrigan can remind me

Reply:
I used `%` because I want brackets in my resulting string